lazyvim/extras/lang/rust.nix

```
@@ -0,0 +1,36 @@
+self:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkEnableOption;
+
+  cfg = config.programs.lazyvim;
+in
+{
+  options.programs.lazyvim.extras.lang.rust = {
+    enable = mkEnableOption "the lang.rust extra";
+  };
+
+  config = mkIf cfg.extras.lang.rust.enable {
+    programs.neovim = {
+      extraPackages = [ pkgs.rust-analyzer ];
+
+      plugins =
+        [
+          (pkgs.vimPlugins.nvim-treesitter.withPlugins (
+            plugins: builtins.attrValues { inherit (plugins) rust; }
+          ))
```

This needs to defin `__darwinAllowLocalNetworking` in order to work on Darwin.

```suggestion
        # NOTE: enabling rust treesitter on Darwin leads to
        # error: The option `programs.neovim.plugins."[definition 2-entry 1]"
        # .__darwinAllowLocalNetworking' does not exist.
        (if pkgs.stdenv.isDarwin then
          pkgs.vimUtils.buildVimPlugin {
            inherit (nvim-treesitter.withPlugins (
              plugins: with plugins; [ rust ron ]
            )) pname version src meta;
            __darwinAllowLocalNetworking = true;
          }
        else
          nvim-treesitter.withPlugins (
            plugins: with plugins; [ rust ron ]
          )
        )
```

 It may also work if the attribute is just defined with an empty string value.

---

If you want to adopt something like this, maybe postpone my alternate list construction from https://github.com/matadaniel/LazyVim-module/pull/6#discussion_r1965932060 since it's inconsistent with the rest of the repo.